### PR TITLE
Tudor/performance fixes

### PR DESCRIPTION
**Pull request by tudor-malene**
### Why this change is needed

We have performance issues because indexes and primary keys are over large byte arrays.

### What changes were made as part of this PR

- change the pk of the batch table to the sequencerId 
- store all searcheable and indexable hashes as byte[16] (there is enough entropy in 128 bits and it should cut the memory usage in half)
- add cache for retreiving an batch by index ( and also rework the caching a bit)

### PR checks pre-merging

Please indicate below by ticking the checkbox that you have read and performed the required
[PR checks](https://github.com/obscuronet/obscuro-internal/blob/main/dev-ops-docs/dev-pr-checks.md)

- [ ] PR checks reviewed and performed

### CODE REVIEW

File: go/enclave/storage/init/edgelessdb/001_init.sql [hunk 1]
```diff
@@ -29,24 +29,21 @@ GRANT ALL ON obsdb.attestation_key TO obscuro;
 
 create table if not exists obsdb.block
 (
-    hash         binary(32),
-    parent       binary(32),
+    hash         binary(16),
```
Comment (BedrockSquirrel) [inline go/enclave/storage/init/edgelessdb/001_init.sql:32]: Is this safe? Likes does ethereum have hash collisions if you truncate the block hashes to 16bytes and what would happen with this table if there was a collision?

Comment (BedrockSquirrel) [reply go/enclave/storage/init/edgelessdb/001_init.sql:32]: Though SHA-256 nominally offers a 256-bit output, no weakness about it is known when the output is truncated to 128 bits, except, of course, weaknesses inherent to the shorter output length; e.g. collision resistance drops from the infeasible 2128 to the possible (but hard) 264.

And it sounds like keccak is similarly secure for truncation. So I guess it's all good?

**Review by BedrockSquirrel — APPROVED** [review]
LGTM so long as we don't think hash collisions would be an issue.

File: go/enclave/storage/init/edgelessdb/001_init.sql (hunk 1 above)
Comment (tudor-malene) [reply go/enclave/storage/init/edgelessdb/001_init.sql:32]: yeah. We'll add checks that the value retrieved from the db hashes to what you expect.
We would have done that anyway as the db must be considered to be vulnerable

**Pull request merged**